astropy/stats/circstats.py

```diff
@@ -0,0 +1,481 @@
+The Maximum Likelihood Estimator for the Von Mises distribution along with the
+Cramer-Rao Lower Bounds are also implemented. Almost all of the implementations
+are based on reference [1], which is also the basis for the R package
+'CircStats'.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+
+
+__all__ = ['circmean','circvar', 'circmoment', 'circcorrcoef', 'rayleightest',
+           'vtest', 'vonmisesmle', 'vonmisescrlb']
+__doctest_requires__ = {'vtest': ['scipy.stats'],
+                        'vonmisescrlb': ['scipy.special']}
+
+
+def _components(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized rectangular components
+    # of the circular data.
+    data = np.asarray(data)
+
+    assert type(p) is int
+    if w is None:
+        w = np.ones(data.shape)
+    else:
+        assert w.shape == data.shape
+
+    C = np.sum(w*np.cos(p*(data-phi)), axis)/np.sum(w, axis)
+    S = np.sum(w*np.sin(p*(data-phi)), axis)/np.sum(w, axis)
+
+    return C,S
+
+def _angle(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized sample mean angle
+    C,S = _components(data,w,p,phi,axis)
+    theta = np.arctan2(S,C)
+
+    mask = (S == .0)*(C == .0)
+    if mask.ndim > 0:
+        theta[mask] = np.nan
+    return theta
+
+def _length(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized sample length
+    C,S = _components(data,w,p,phi,axis)
+    return np.hypot(S,C)
+
+def circmean(data, w=None, axis=None):
+    """ Computes the circular mean angle of an array of circular data.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array of circular (directional) data measured in radians.
+    w : numpy.ndarray, optional
+        In case of grouped data, the i-th element of ``w`` represents a
+        weighting factor for each group such that sum(w,axis) equals the number
+        of observations. See [1], remark 1.4, page 22, for detailed
+        explanation.
+    axis : int, optional
+        Axis along which circular means are computed. The default is to compute
+        the mean of the flattened array.
+
+    Returns
+    -------
+    circmean : float
+        Circular mean.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.stats.circstats import circmean
+    >>> data = np.deg2rad(np.array([51, 67, 40, 109, 31, 358]))
+    >>> circmean(data) # doctest: +FLOAT_CMP
+    0.84870441244501904
+
+    References
+    ----------
+    .. [1] S. R. Jammalamadaka, A. SenGupta. "Topics in Circular Statistics".
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
```

Comment:
@mirca, yes, you are right. In this case, the doc you have is consistent with the package. Never mind. :)